core/flyout.js

```diff
@@ -739,36 +889,67 @@ Blockly.Flyout.prototype.createBlockFunc_ = function(originBlock) {
  * @private
  */
 Blockly.Flyout.prototype.placeNewBlock_ = function(originBlock, workspace) {
-  // Create the new block by cloning the block in the flyout (via XML).
-  var xml = Blockly.Xml.blockToDom(originBlock);
-  var block = Blockly.Xml.domToBlock(xml, workspace);
-  // Place it in the same spot as the flyout copy.
   var svgRootOld = originBlock.getSvgRoot();
   if (!svgRootOld) {
     throw 'originBlock is not rendered.';
   }
+  // Figure out where the original block is on the screen, relative to the upper
+  // left corner of the main workspace.
   var xyOld = Blockly.getSvgXY_(svgRootOld, workspace);
-  // Scale the scroll (getSvgXY_ did not do this).
-  if (this.RTL) {
-    var width = workspace.getMetrics().viewWidth - this.width_;
-    xyOld.x += width / workspace.scale - width;
-  } else {
-    xyOld.x += this.workspace_.scrollX / this.workspace_.scale -
-        this.workspace_.scrollX;
+  // Take into account that the flyout might have been scrolled horizontally
+  // (separately from the main workspace).
+  // Generally a no-op in vertical mode but likely to happen in horizontal
+  // mode.
+  var scrollX = this.workspace_.scrollX;
+  var scale = this.workspace_.scale;
+  xyOld.x += scrollX / scale - scrollX;
+  // If the flyout is on the right side, (0, 0) in the flyout is offset to
+  // the right of (0, 0) in the main workspace.  Offset to take that into
+  // account.
+  if (this.toolboxPosition_ == Blockly.TOOLBOX_AT_RIGHT) {
+    scrollX = workspace.getMetrics().viewWidth - this.width_;
+    scale = workspace.scale;
```

Comment:
what's the difference between workspace_.scale and workspace.scale?  Practically, they should be the same since zooming changes both?  Just want to make sure the sure the different uses of workspace.scale and this.workspace_.scale are intentional both here and when you're calculating Y.  I don't currently understand why the toolbox on the right and toolbox at bottom would want to use a different scale than left and top.